### ISSUE TITLE
[a11y]: ContentSwitcher [all variants] - TabList role needed for each tab group

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#13977]
### Package

@carbon/react

### Browser

FF & Safari & iPhone

### Package version

@carbon/react@1.30.0

### Automated testing tool and ruleset
Manual QA

### Assistive technology

JAWS 

### Description

TabList role for each tab group need

the tablist role does not have accessible label. Hence the SR does not identify the semantics of the tab group

### WCAG 2.1 Violation
1.3.1 Info and Relationships 

### Reproduction/example

Record `Number 47` in the below table:
https://airtable.com/shrFsaeHhImOkuVNR

### Steps to reproduce

When the screen reader is unable to identify a label, it’s difficult for users to understand the purpose of the control.

### Recommended Fix: 
Identify the label for the tablist using aria-label or aria-labelledby. 

Note: in the earlier versions of the Browser or SR the tab group role and the label is not recognized even when its made accessible. Now SR does recognize the semantic. Hence its recommended to add Accessible label.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by heloiselui** [comment]
Checked with our accessibility team, and this issue still exists.
